package-coverage/generator/generator.go

```diff
@@ -78,10 +78,35 @@ type Generator struct {
 
 	// Tags is arguments passed to the go test runner
 	Tags string
+
+	// SequentialMode controls if the tests are run in parallel or not.  Original version was sequential
+	SequentialMode bool
 }
 
 func (g *Generator) do(paths []string) {
-	for _, path := range paths {
-		generateCoverage(path, g.Exclusion, g.QuietMode, g.Tags)
+	if g.SequentialMode {
+		for _, path := range paths {
+			generateCoverage(path, g.Exclusion, g.QuietMode, g.Tags)
+		}
+	} else {
+		resultCh := make(chan struct{})
+		defer close(resultCh)
+
+		// run in parallel
+		for _, path := range paths {
+			go func(inPath string, inExclusion *regexp.Regexp, inQuietMode bool, inTags string) {
+				generateCoverage(inPath, inExclusion, inQuietMode, inTags)
+				resultCh <- struct{}{}
+			}(path, g.Exclusion, g.QuietMode, g.Tags)
+		}
+
+		// wait until everything is done
+		done := 0
```

Comment:
how about using `sync.WaitGroup` instead?

Reply:
Was my first thought but decided on this style to allow me to select on the channel read (and potentially) timeout.
Then I realised that I didn't have support for timeouts yet.

I think if I use the concurrency approach, this will preclude `sync.WaitGroup`